synapse/storage/databases/main/room.py

```diff
@@ -1540,14 +1546,24 @@ def _get_event_reports_paginate_txn(txn):
 
             args += [limit, start]
             txn.execute(sql, args)
-            event_reports = self.db_pool.cursor_to_dict(txn)
 
+            event_reports = []
             if count > 0:
```

Comment:
This `if` probably isn't necessary (assuming our DB code isn't buggy). `event_reports` will still end up as an empty list.